consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/IbftGossip.java

```diff
@@ -100,9 +101,16 @@ public boolean gossipMessage(final Message message) {
       final List<Address> excludeAddressesList =
           Lists.newArrayList(
               message.getConnection().getPeer().getAddress(), signedData.getSender());
-      multicaster.send(messageData, excludeAddressesList);
-      seenMessages.add(signature);
+      transmit(messageData, signature, excludeAddressesList);
       return true;
     }
   }
+
+  public void transmit(
```

Comment:
By doing this, you're exposing a way of transmission that bypasses the `gossipMessage()` check that ensures seen messages are not repeated. Where this is called, does it hold true that there can never be duplicate invocations?

Reply:
Made this slightly nicer, but making the Hash of MessageData the unique key ... and then hiding the "was this sent" mentality inside the send function.